internal/isoeditor/rhcos.go

```diff
@@ -115,14 +127,36 @@ func (e *rhcosEditor) embedInitrdPlaceholders() error {
 	return nil
 }
 
+func (e *rhcosEditor) embedOffsetsInSystemArea(isoPath string) error {
+	ignitionOffset, err := isoutil.GetFileLocation(ignitionImagePath, isoPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get ignition image offset")
+	}
+
+	ramDiskOffset, err := isoutil.GetFileLocation(ramDiskImagePath, isoPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get ram disk image offset")
+	}
+
+	var ignitionOffsetInfo isoutil.OffsetInfo
+	copy(ignitionOffsetInfo.Key[:], ignitionHeaderKey)
+	ignitionOffsetInfo.Offset = ignitionOffset
```

Comment:
The original implementation of these also included a length.
ref: https://github.com/openshift/assisted-service/blob/cdd3806e2b8d3e0195e7588135992e56ab441c3a/pkg/s3wrapper/upload_iso.go#L180-L181

Can we make this compatible so that we can share parsing code between the full and minimal ISO in the future?

In fact, @djzager is working on this currently for the full iso I think.

Reply:
I have my, currently broken, implementation of embedding ignition in https://github.com/openshift/assisted-service/pull/1177 the big difference I saw when I reviewed this PR is (as I understand it) that I'm grabbing the info directly from the iso versus the extracted iso.

Reply:
Added Length to OffsetInfo struct and used unit64, so it should now adhere the format used in coreos-assembler [*] I.e. '8s2Q' - [8]byte uint64 uint64

[*] https://github.com/coreos/coreos-assembler/blob/510dbec7b84aa45a646079fe3341e2b5925c0774/src/cmd-buildextend-live#L510